src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1760,14 +1771,25 @@ ContainsUnionSubquery(Query *queryTree)
  * index that the partition key of the relation exists in the query. The query is
  * found in the planner info of the relation restriction.
  */
-static Index
-RelationRestrictionPartitionKeyIndex(RelationRestriction *relationRestriction)
+static Var *
+RelationRestrictionPartitionKeyIndex(Query *originalQuery,
+									 RelationRestriction *relationRestriction,
+									 Index *partitionKeyIndex)
 {
 	ListCell *targetEntryCell = NULL;
 	Index partitionKeyTargetAttrIndex = 0;
 
-	PlannerInfo *relationPlannerRoot = relationRestriction->plannerInfo;
-	Query *relationPlannerParseQuery = relationPlannerRoot->parse;
+
+	FindRteIdentityContext *findRteIdentityContext = palloc0(
+		sizeof(FindRteIdentityContext));
+	findRteIdentityContext->targetRTEIdentity = GetRTEIdentity(relationRestriction->rte);
+	FindQueryContainingRTEIdentity((Node *) originalQuery, findRteIdentityContext);
+	if (!findRteIdentityContext->found)
+	{
+		return NULL;
```

Comment:
Is it ever possible that we cannot find the query? I'm fine with having this check, but might use a comment that we should always find the query

Reply:
Assuming we don't have a bug in the `FindQueryContainingRTEIdentity` function, we should find the query, however I think it is good to have this check because if we have a bug etc then instead of crashing, we will just recursively plan a query, which might be pushdownable or not.